tests/util/jsx.js

```diff
@@ -71,11 +69,27 @@ describe('jsxUtil', () => {
     });
 
     it('Can ignore null', () => {
-      assertInValid(`
+      assertValid(`
         function Test() {
           return null;
         }
       `);
     });
+
+    it('Ignores JSX arguments to function calls used as return value of arrow functions', () => {
+      let astNode = parseCode(`const obj = {
+        prop: () => test(<a>something</a>)
+      }`);
+      let arrowFunctionExpression = astNode.declarations[0].init.properties[0].value;
+
+      assert(!isReturningJSX(() => false, arrowFunctionExpression, mockContext));
```

Comment:
maybe these could use the `assertValid`/`assertInvalid` helpers?

Reply:
I don't think I can. The node passed to `isReturningJSX` needs to be the `ArrowFunctionExpression` to illustrate the issue. But in the examples I provided, it ends up being the `AssignmentExpression` if I try to use `assertValid` as-is IIRC. And just putting in the arrow function by itself passes a `FunctionExpressionStatement` or something like that. It doesn't match what's being passed from `Components.js`, in short

Reply:
You can't pass just `() => test(<a>something</a>)`? or are you saying it needs the context provided by being inside the object literal

Reply:
Yeah that context is relevant. In the case of the objects it's the being in the `Property` context and in the second example it's the being in the `AssignmentExpression` context